core/src/test/java/ai/timefold/solver/core/impl/localsearch/decider/acceptor/restart/UnimprovedMoveCountStuckCriterionTest.java

```diff
@@ -15,106 +17,108 @@
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
-class UnimprovedMoveCountRestartStrategyTest {
+class UnimprovedMoveCountStuckCriterionTest {
 
     @Test
-    void isTriggered() {
+    void isSolverStuck() {
         var clock = mock(Clock.class);
```

Comment:
Instead of mocking the Clock, you can actually get a new instance which you can control to do literally anything you like.

Reply:
I need to create a mock and ensure we can test the logic after the grace period.

Reply:
But isn't `Clock` designed for that?

You can create `Clock` instances based on any `Instant` you want. You can store the clock with that instant in the class, and the class will immediately use that time value. 

Or you can configure a clock that ticks per a certain value, and have time move every 5 minutes or something like that. IMO mocking the Clock is not how the class is supposed to be used.